plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -126,6 +148,31 @@ findPosition pos root = go Nothing root
             startOfRight <- _start . _codeRange_range <$> V.headM right
             if pos < startOfRight then binarySearchPos left else binarySearchPos right
 
+-- | Traverses through the code range and children to a folding ranges
+findFoldingRanges :: CodeRange -> [FoldingRange]
+findFoldingRanges r@(CodeRange _ children _) =
+  let frRoot :: [FoldingRange] = case createFoldingRange r of
+        Just x  -> [x]
+        Nothing -> []
+
+      frChildren :: [FoldingRange] = concat $ V.toList $ fmap findFoldingRanges children
+   in frRoot ++ frChildren
+
+-- | Parses code range to folding range
+createFoldingRange :: CodeRange -> Maybe FoldingRange
+createFoldingRange node1 = do
+    let range = _codeRange_range node1
+    let Range startPos endPos = range
+    let Position lineStart _= startPos
+    let Position lineEnd _ = endPos
+    let codeRangeKind = _codeRange_kind node1
```

Comment:
```suggestion
createFoldingRange (CodeRange (Range (Position lineStart _) (Position lineEnd _) _ codeRangeKind)) = do
```

Reply:
Looks a lot cleaner this way. I've changed it.